src/main/java/com/kelvinconnect/discord/command/JoinLeaveCommand.java

```diff
@@ -40,27 +58,78 @@ private static void debugPrintChannels(Server server) {
     }
 
     private void initChannels() {
-        channels = Arrays.asList(new KCChannel(276318041443270657L, 763082445524041761L, "pubchat"),
-                new KCChannel(689418031151054944L, 763085327270543381L, "corona", "covid"),
-                new KCChannel(550614902532997131L, 763085269435678791L, "spamdo"),
-                new KCChannel(421223895513956352L, 421225268057735168L, "music"),
-                new KCChannel(694117175392469032L, 763083545451167774L, "film", "tv"),
-                new KCChannel(408933031064371200L, 763083371193696267L, "roll", "role"),
-                new KCChannel(425314799455436801L, 425314649458737163L, "anime", "weeb"),
-                new KCChannel(763082945979744278L, 763082196985970709L, "pets"),
-                new KCChannel(770330765133086751L, 770330666411360299L, "food"),
-                new KCChannel(763091730605408307L, 763091623386021888L, "programming"),
-                new KCChannel(763091960541478912L, 763091847559118868L, "botdev", "kcbot"),
-                new KCChannel(763083029794390017L, 763082257429692458L, "tech"),
-                new KCChannel(763083100594110524L, 763082324748533790L, "sport"),
-                new KCChannel(763086481560305704L, 763086365835395123L, "politics"),
-                new KCChannel(421623339145232404L, 421623272434565130L, "games"),
-                new KCChannel(763089542152323112L, 763089285632884756L, "tetris"),
-                new KCChannel(753587144991178792L, 753586719999393872L, "lol"),
-                new KCChannel(763082784784515092L, 763082509952876546L, "wow"),
-                new KCChannel(874359730540253204L, 874359350209151046L, "ffxiv", "ff14"));
+        try {
+            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+            DocumentBuilder documentBuilder = dbf.newDocumentBuilder();
+            URLConnection urlConnection = new URL(KC_CHANNEL_LIST_URL).openConnection();
+            urlConnection.addRequestProperty("Accept", "application/xml");
+            Document channelListDocument = documentBuilder.parse(urlConnection.getInputStream());
```

Comment:
Would it not be better to load the channels from a local file rather than from whatever's committed on GitHub? The point of this feature is so we can add channels without needing to make changes to the source code.

Reply:
My thinking was that an xml file on github can be modified by someone with write access to the repo from _anywhere,_ rather than having to remote in to the server the bot is running on to edit the file. The file can of course be held anywhere if we load from URL, and leaves the door open for cloud hosting in the future.

Reply:
What about loading the file from a URI that can be passed in via cmd line arguments? That way we can load from a file _or_ from a web URL.